libs/fluid-elements/tab-group/src/lib/tab-group.ts

```diff
@@ -107,38 +122,58 @@ export class FluidTabGroup extends LitElement {
         index = this.tabChildren.length - 1;
       }
 
+      // Adds a class that activates the outline when focused
+      this.tabChildren[index].tabbed = true;
       this.tabChildren[index].focus();
-      this.tabChildren[index].tabindex = 0;
-      this.tabChildren[oldIndex].tabindex = -1;
-      this.activetabid = this.tabChildren[index].tabid;
+      this._toBeActiveTab = this.tabChildren[index].tabid;
+      // Set the old to normal state
+      this.tabChildren[oldIndex].tabbed = false;
+    }
+  }
+
+  /** In order to prevent the browser to scroll when the user selects a tab using the spacebar we prevent the default behavior. */
+  private handleKeyDown(event: KeyboardEvent): void {
+    if (event.code === SPACE) {
+      event.preventDefault();
     }
   }
 
-  /** Checks whether the next tab is also disabled or not and sets the next not disabled tab as active  */
+  /** Checks whether the next tab is also disabled or not and sets the next available tab as active  */
   private handleDisabled(disableTabEvent: FluidTabDisabledEvent): void {
-    if (this.activetabid === disableTabEvent.disableTab) {
-      const tabToEnable = this.tabChildren.find((tab) => !tab.disabled);
-      if (tabToEnable) {
-        tabToEnable.active = true;
-      }
+    if (this.activeTabId === disableTabEvent.disableTab) {
+      this.setFirstEnabledTabActive();
     }
   }
 
   /** Handles changes in the slot. Initially sets the active item (default is first) */
   private slotchange(): void {
+    // Contains the length of the tabChildren to check if the user has removed tabs
+    let tabChildrenLength;
+    if (this.tabChildren !== undefined) {
+      tabChildrenLength = this.tabChildren.length;
+    }
     this.tabChildren = Array.from(this.querySelectorAll('fluid-tab'));
-    // Initially set the first tab to active
-    if (!this.activetabid && this.tabChildren.length > 0) {
+    // Check if one of the tabs is focusable and if the user removed tabs from the tabs Array if non are focusable
+    // If so we set the first available (not disabled) tab active
+    if (
+      !this.tabChildren.find((tab) => tab.tabindex === 0) &&
+      tabChildrenLength > this.tabChildren.length
+    ) {
+      this.setFirstEnabledTabActive();
+    }
```

Comment:
This feels like something that could be some shareable logic with what happens in L165-L176
Maybe this could be simplified.